packages/dx-react-chart/src/plugins/zoom-pan.tsx

```diff
@@ -97,47 +112,39 @@ class ZoomAndPanBase extends React.PureComponent<ZoomAndPanProps, ZoomAndPanStat
 
   handleStart(zoomRegionKey: string, e: any) {
     this.offset = getOffset(e.currentTarget);
+    const coords = getCoordsWithOffset(e, this.offset);
       // Rectangle mode should be canceled if "zoomRegionKey" is released during mouse movevent or
       // not pressed when mouse is up. To do it access to "event" object is required in
-      // "handleMouseMove" and "handleMouseUp".
+      // "handleMove" and "handleEnd".
       // TODO: Provide rectangle mode canceling.
     if (isKeyPressed(e, zoomRegionKey)) {
-      this.rectOrigin = [e.pageX - this.offset[0], e.pageY - this.offset[1]];
+      this.rectOrigin = coords;
     }
     if (e.touches && e.touches.length === 2) {
       this.multiTouchDelta = getDeltaForTouches(e.touches).delta;
     }
+    this.lastCoordinates = coords;
   }
 
-  handleTouchMove(scales: ScalesCache, e: any) {
+  handleMove(scales: ScalesCache, e: any) {
     e.preventDefault();
     if (e.touches && e.touches.length === 2) {
       const current = getDeltaForTouches(e.touches);
       this.zoom(scales, current.delta - this.multiTouchDelta!, current.center);
       this.multiTouchDelta = current.delta;
     } else {
-      this.handleMouseMove(scales, { x: e.touches[0].clientX, y: e.touches[0].clientY });
+      this.scroll(scales, e.touches ? e.touches[0] : e);
```

Comment:
The `e.touches ? e.touches[0] : e` code is the same as that in event coordinates calculation. Can't it be replaced with this?
```jsx
this.scroll(scales, getEventCoords(e));
```

Reply:
No, there is no `getBoundingClientRect` in event.